### PR TITLE
Fix-up scripts so that external builds pass

### PR DESCRIPTION
**Pull request by SerialVelocity**
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/1319)

### CODE REVIEW

File: scripts/circle-ci/install-docker.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+set -x
+
+MIN_VERSION_WANTED=1.10.0
+CURRENT_VERSION=$(docker version --format '{{.Client.Version}}')
+
+if [ ! "$MIN_VERSION_WANTED" = "$(echo -e "$CURRENT_VERSION\n$MIN_VERSION_WANTED" | sort -V | head -n 1)" ]; then
```
Comment (jeremyk-91) [inline scripts/circle-ci/install-docker.sh]: nit: same as `!=`?

File: scripts/circle-ci/pr-changelog-status-check.sh [hunk 2]
```diff
@@ -4,23 +4,39 @@ set -x
...
 CURRENT_REF=$(git log -1 --format="%H")
 CHANGELOG_COMMIT_FLAG=$(git log --pretty=format:%B origin/develop..HEAD | grep -iq '\[no release notes\]')$?
 CHANGELOG_MODIFIED=$(git log --name-only --pretty=format: origin/develop..HEAD | grep -q $CHANGELOG)$?
 
+success() {
+    message=$1
+    if [ -z "$GITHUB_AUTH_TOKEN" ]; then
+        echo "$message"
+    else
+        curl -X POST -f --silent --header "Authorization: token $GITHUB_AUTH_TOKEN" -d '{"state": "success", "description": "'"$message"'", "context": "changelog"}' "https://api.github.com/repos/palantir/atlasdb/statuses/$CURRENT_REF" >/dev/null
```
Comment (jeremyk-91) [inline scripts/circle-ci/pr-changelog-status-check.sh]: OK - I have verified that this (changed in commit 3) string-substitution works as expected.

Comment (SerialVelocity) [inline scripts/circle-ci/pr-changelog-status-check.sh]: If you look below, it shows you in the checks :)

File: scripts/circle-ci/install-docker.sh (hunk 1 above)
Comment (SerialVelocity) [inline scripts/circle-ci/install-docker.sh]: I'll use `-ne` instead since `!=` has different semantics in double brackets

**Pull request merged**